src/desktop/apps/auth2/components/__tests__/ModalContainer.jest.js

```diff
@@ -19,28 +19,37 @@ describe('ModalContainer', () => {
     sd.CSRF_TOKEN = 'sample-token'
   })
 
-  it('Mediator can open a login modal', () => {
+  it('Mediator can open a login modal', done => {
     const component = mount(<ModalContainer />)
     mediator.trigger('open:auth', { mode: 'login' })
-    const form = component.find(ModalManager).instance().state
 
-    expect(form.currentType).toBe('login')
+    setTimeout(() => {
+      const form = component.find(ModalManager).instance().state
+      expect(form.currentType).toBe('login')
+      done()
+    }, 800)
   })
 
-  it('Mediator can open a signup modal', () => {
+  it('Mediator can open a signup modal', done => {
     const component = mount(<ModalContainer />)
     mediator.trigger('open:auth', { mode: 'register' })
-    const form = component.find(ModalManager).instance().state
 
-    expect(form.currentType).toBe('signup')
+    setTimeout(() => {
+      const form = component.find(ModalManager).instance().state
+      expect(form.currentType).toBe('signup')
+      done()
+    }, 800)
   })
 
-  it('Mediator can open a reset_password modal', () => {
+  it('Mediator can open a reset_password modal', done => {
     const component = mount(<ModalContainer />)
     mediator.trigger('open:auth', { mode: 'reset_password' })
-    const form = component.find(ModalManager).instance().state
 
-    expect(form.currentType).toBe('reset_password')
+    setTimeout(() => {
+      const form = component.find(ModalManager).instance().state
+      expect(form.currentType).toBe('reset_password')
+      done()
+    }, 800)
```

Comment:
Ah I'm a little worried about these tests since it adds to the overall testing time. If it's because of the `document.readyState`, then we can mock setTimeout and run fake timers to avoid it.

Reply:
Here's an example that might help: https://github.com/artsy/reaction/blob/b566045d54531e51ff33769c14a96fd445862937/src/Components/Publishing/Video/Player/__tests__/VideoPlayer.test.tsx#L13

Reply:
sure, ok i'll do that instead